homeassistant/components/unifi/device_tracker.py

```diff
@@ -62,111 +60,122 @@
-
-        except aiounifi.LoginRequired:
-            try:
-                with async_timeout.timeout(5):
-                    await self.controller.login()
-            except (asyncio.TimeoutError, aiounifi.AiounifiException):
-                clients = []
-
-        except aiounifi.AiounifiException:
-            clients = []
-
-        # Filter clients to provided SSID list
-        if self._ssid_filter:
-            clients = [
-                client for client in clients
-                if client.essid in self._ssid_filter
-            ]
-
-        self._clients = {
-            client.raw['mac']: client.raw
-            for client in clients
-            if (dt_util.utcnow() - dt_util.utc_from_timestamp(float(
-                client.last_seen))) < self._detection_time
-        }
+        """Synchronize state with controller."""
+        await self.controller.request_update()
 
-    async def async_scan_devices(self):
-        """Scan for devices."""
-        await self.async_update()
-        return self._clients.keys()
+    @property
+    def is_connected(self):
+        """Return true if the device is connected to the network."""
+        detection_time = self.controller.unifi_config[CONF_DETECTION_TIME]
 
-    def get_device_name(self, device):
-        """Return the name (if known) of the device.
+        if (dt_util.utcnow() - dt_util.utc_from_timestamp(float(
+                self.client.last_seen))) < detection_time:
+            return True
+        return False
 
-        If a name has been set in Unifi, then return that, else
-        return the hostname if it has been detected.
-        """
-        client = self._clients.get(device, {})
-        name = client.get('name') or client.get('hostname')
-        _LOGGER.debug("Device mac %s name %s", device, name)
-        return name
+    @property
+    def source_type(self):
+        """Return the source type, eg gps or router, of the device."""
+        return SOURCE_TYPE_ROUTER
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        return self.client.name or self.client.hostname
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this switch."""
+        return 'unifi-dt-{}'.format(self.client.mac)
+
+    @property
+    def available(self) -> bool:
+        """Return if controller is available."""
+        return self.controller.available
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+        return {
+            'connections': {(CONNECTION_NETWORK_MAC, self.client.mac)}
+        }
 
-    def get_extra_attributes(self, device):
+    @property
+    def device_state_attributes(self):
         """Return the extra attributes of the device."""
-        if not self._monitored_conditions:
-            return {}
+        monitored_conditions = \
+            self.controller.unifi_config.get(CONF_MONITORED_CONDITIONS, {})
```

Comment:
I'm assuming this is in agreement with https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md. Ie, that there is a reason like this:
> Integrations should only include selector logic if it make sense in the context of interface, like if it would require extra requests.

Reply:
This is merely to keep with the current implementation. I'm all for removing them and adding them back in config entry options later on

Reply:
I'm not sure ARD0003 allows config entry options for monitored conditions. I'm reading it like we don't want any options for this type at all.

Reply:
Ok, removing it. People can bring up a discussion if they want specific things back